test/integration/element_tracing.cc

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <iostream>
+#include <string>
+#include <gtest/gtest.h>
+
+#include "sdf/SDFImpl.hh"
+#include "sdf/parser.hh"
+#include "sdf/Model.hh"
+#include "sdf/Joint.hh"
+#include "sdf/Link.hh"
+#include "sdf/Root.hh"
+#include "sdf/World.hh"
+#include "sdf/Actor.hh"
+#include "sdf/Light.hh"
+#include "test_config.h"
+
+//////////////////////////////////////////////////
+std::string findFileCb(const std::string &_input)
+{
+  return sdf::testing::TestFile("integration", "model", _input);
+}
+
+//////////////////////////////////////////////////
+void checkElementPtr(const sdf::ElementPtr &_elem,
```

Comment:
I see why this was made into a function, since you are calling the same tests repeatedly, but this makes it difficult to debug when a test fails. For example, here is the output of the failing windows CI:

<details>
<summary>Expand</summary>

```
17: C:\Jenkins\workspace\sdformat-ci-pr_any-windows7-amd64\ws\sdformat\test\integration\element_tracing.cc(44): error: Expected equality of these values:
17:   _filePath
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_actor/model.sdf"
17:   _elem->FilePath()
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_actor\\model.sdf"
17: C:\Jenkins\workspace\sdformat-ci-pr_any-windows7-amd64\ws\sdformat\test\integration\element_tracing.cc(44): error: Expected equality of these values:
17:   _filePath
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_actor/model.sdf"
17:   _elem->FilePath()
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_actor\\model.sdf"
17: C:\Jenkins\workspace\sdformat-ci-pr_any-windows7-amd64\ws\sdformat\test\integration\element_tracing.cc(44): error: Expected equality of these values:
17:   _filePath
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_light/model.sdf"
17:   _elem->FilePath()
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_light\\model.sdf"
17: C:\Jenkins\workspace\sdformat-ci-pr_any-windows7-amd64\ws\sdformat\test\integration\element_tracing.cc(44): error: Expected equality of these values:
17:   _filePath
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_light/model.sdf"
17:   _elem->FilePath()
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_light\\model.sdf"
17: C:\Jenkins\workspace\sdformat-ci-pr_any-windows7-amd64\ws\sdformat\test\integration\element_tracing.cc(44): error: Expected equality of these values:
17:   _filePath
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_model/model.sdf"
17:   _elem->FilePath()
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_model\\model.sdf"
17: C:\Jenkins\workspace\sdformat-ci-pr_any-windows7-amd64\ws\sdformat\test\integration\element_tracing.cc(44): error: Expected equality of these values:
17:   _filePath
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_model/model.sdf"
17:   _elem->FilePath()
17:     Which is: "C:/Jenkins/workspace/sdformat-ci-pr_any-windows7-amd64/ws/sdformat\\test\\integration\\model\\test_model\\model.sdf"
17: [  FAILED  ] ElementTracing.includes (196 ms)
17: [----------] 2 tests from ElementTracing (252 ms total)
17: 
17: [----------] Global test environment tear-down
17: [==========] 2 tests from 1 test case ran. (252 ms total)
17: [  PASSED  ] 1 test.
17: [  FAILED  ] 1 test, listed below:
17: [  FAILED  ] ElementTracing.includes
17: 
17:  1 FAILED TEST
 17/198 Test  #17: INTEGRATION_element_tracing ............................***Failed    0.31 sec
```

</details>

All the errors point to line 44 so it's difficult to see what is actually failing

Reply:
Yup, that makes a whole lot of sense. I was running `gdb` the whole time, and didn't notice it since I was debugging `parser.cc`. Thanks for catching that! https://github.com/osrf/sdformat/pull/548/commits/ee7f583a830334d39adfc9a2443fab96ac433b11